supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/statuspolling/JesApiQueryManager.scala

```diff
@@ -82,8 +102,28 @@ class JesApiQueryManager(val qps: Int Refined Positive) extends Actor with Actor
   private case class BeheadedWorkQueue(workToDo: Option[NonEmptyList[JesApiQuery]], newWorkQueue: Queue[JesApiQuery])
   private def beheadWorkQueue(maxBatchSize: Int): BeheadedWorkQueue = {
 
-    val head = workQueue.take(maxBatchSize).toList
-    val tail = workQueue.drop(maxBatchSize)
+    /*
+      * Take the head of the queue, making sure it stays under maxBatchSize as well as maxBatchRequestSize.
+      * Assumes that each query in the queue can fit in an empty batch (queries with a size > maxBatchSize should no be added to the queue)
+      *
+      * This is a naive approach where we keep adding queries to the head as long as they don't overflow it, and then we stop.
+      * This could leave us with half empty batches and is not very efficient.
+      *
+      * A somewhat less naive approach would be to keep looking in the queue for another query that would fit.
+      *
+      * More generally this is the knapsack problem (https://en.wikipedia.org/wiki/Knapsack_problem) which is NP-Complete.
+      * If this needs further optimization, finding an approximation algorithm that fits well this case would be a place to start.
+      */
+    @tailrec
+    def behead(queue: Queue[JesApiQuery], head: Vector[JesApiQuery]): Vector[JesApiQuery]  = queue.headOption match {
+      // TODO: don't recompute the length sum every time ?
+      case Some(query) if head.size < maxBatchSize && head.map(_.contentLength).sum + query.contentLength < maxBatchRequestSize =>
+        behead(queue.tail, head :+ query)
+      case _ => head
+    }
+
+    val head = behead(workQueue, Vector.empty).toList
```

Comment:
Yeah, TBH the reason I had the actor reject the request if it was > 15MB was to avoid having to deal with this here in the first place but I guess we can do both.